plugins/playlist/src/components/PlaylistPage/PlaylistEntitiesTable.tsx

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Entity,
+  parseEntityRef,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import {
+  ErrorPanel,
+  SubvalueCell,
+  Table,
+  TableFilter,
+} from '@backstage/core-components';
+import { errorApiRef, useApi } from '@backstage/core-plugin-api';
+import { catalogApiRef, EntityRefLink } from '@backstage/plugin-catalog-react';
+import { usePermission } from '@backstage/plugin-permission-react';
+import { permissions } from '@backstage/plugin-playlist-common';
+import AddBoxIcon from '@material-ui/icons/AddBox';
+import DeleteIcon from '@material-ui/icons/Delete';
+import SearchIcon from '@material-ui/icons/Search';
+import React, { forwardRef, useCallback, useEffect, useState } from 'react';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+
+import { playlistApiRef } from '../../api';
+import { AddEntitiesDrawer } from './AddEntitiesDrawer';
+
+export const PlaylistEntitiesTable = ({
+  playlistId,
+}: {
+  playlistId: string;
+}) => {
+  const catalogApi = useApi(catalogApiRef);
+  const errorApi = useApi(errorApiRef);
+  const playlistApi = useApi(playlistApiRef);
+  const [openAddEntitiesDrawer, setOpenAddEntitiesDrawer] = useState(false);
+
+  const { allowed: editAllowed } = usePermission({
+    permission: permissions.playlistListUpdate,
+    resourceRef: playlistId,
+  });
+
+  const [{ value: entities, loading, error }, loadEntities] =
+    useAsyncFn(async () => {
+      const entityRefs = await playlistApi.getPlaylistEntities(playlistId);
+      if (!entityRefs.length) {
+        return [];
+      }
+
+      const filter = entityRefs.map(ref => {
+        const compoundRef = parseEntityRef(ref);
+        return {
+          kind: compoundRef.kind,
+          'metadata.namespace': compoundRef.namespace,
+          'metadata.name': compoundRef.name,
+        };
+      });
+
+      // TODO(kuanpg): entities in this playlist that no longer exist in the catalog will be
+      // excluded from this response, we need a way to clean up these orphaned refs potentially
+      // via catalog events (https://github.com/backstage/backstage/issues/8219)
```

Comment:
Another thing that might help is #12254. Could also be better (and faster) to move this to the backend? Think that should solve the catalog permission issue too.